### PR TITLE
Support getUnderFSType for ozone,cosn,cephfs-hadoop

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes are proposed in this pull request?

Support getUnderFSType for ozone,cosn,cephfs-hadoop which extends hdfs underfilesystem ufs.

### Why are the changes needed?

image

### Does this PR introduce any user facing changes?

Please list the user-facing changes introduced by your change, including
  1. change in user-facing APIs
  2. addition or removal of property keys
  3. webui

### CODE REVIEW

**Comment by maobaolong** [comment]
@jiacheliu3 Would you like to take a look at this PR?

**Review by jiacheliu3 — APPROVED** [review]
LGTM, thanks for the improvement!

File: underfs/cephfs-hadoop/src/main/java/alluxio/underfs/cephfshadoop/CephfsHadoopUnderFileSystemFactory.java [hunk 1]
```diff
@@ -11,21 +11,31 @@
...
+
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * Factory for creating {@link HdfsUnderFileSystem}.
+ * Factory for creating {@link CephfsHadoopUnderFileSystem}.
  *
- * It caches created {@link HdfsUnderFileSystem}s, using the scheme and authority pair as the key.
+ * It caches created {@link CephfsHadoopUnderFileSystem}s,
+ * using the scheme and authority pair as the key.
```
Comment (dbw9580) [inline underfs/cephfs-hadoop/src/main/java/alluxio/underfs/cephfshadoop/CephfsHadoopUnderFileSystemFactory.java:28]: This is no longer the case. Please remove the misleading comment, and I'll create another PR cleaning up the comment inherited from HDFS UFS factory.

File: underfs/cosn/src/main/java/alluxio/underfs/cosn/CosNUnderFileSystemFactory.java [hunk 2]
```diff
@@ -11,21 +11,30 @@
...
+import com.google.common.base.Preconditions;
+
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * Factory for creating {@link HdfsUnderFileSystem}.
+ * Factory for creating {@link CosnUnderFileSystem}.
  *
- * It caches created {@link HdfsUnderFileSystem}s, using the scheme and authority pair as the key.
+ * It caches created {@link CosnUnderFileSystem}s, using the scheme and authority pair as the key.
```
Comment (dbw9580) [inline underfs/cosn/src/main/java/alluxio/underfs/cosn/CosNUnderFileSystemFactory.java:27]: same here

File: underfs/cosn/src/main/java/alluxio/underfs/cosn/CosnUnderFileSystem.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+   * @return a new Cosn {@link UnderFileSystem} instance
+   */
+  public static CosnUnderFileSystem createInstance(AlluxioURI ufsUri,
+      UnderFileSystemConfiguration conf) {
+    Configuration hdfsConf = createConfiguration(conf);
+    return new CosnUnderFileSystem(ufsUri, conf, hdfsConf);
+  }
+
+  /**
+   * Constructs a new HDFS {@link UnderFileSystem}.
```
Comment (dbw9580) [inline underfs/cosn/src/main/java/alluxio/underfs/cosn/CosnUnderFileSystem.java:43]:
```suggestion
   * Constructs a new COSN {@link UnderFileSystem}.
```

File: underfs/cosn/src/main/java/alluxio/underfs/cosn/CosnUnderFileSystem.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+   *
+   * @param ufsUri the {@link AlluxioURI} for this UFS
+   * @param conf the configuration for this UFS
+   * @param hdfsConf the configuration for HDFS
```
Comment (dbw9580) [inline underfs/cosn/src/main/java/alluxio/underfs/cosn/CosnUnderFileSystem.java:47]:
```suggestion
   * @param hdfsConf the configuration for COSN
```

File: underfs/cephfs-hadoop/src/main/java/alluxio/underfs/cephfshadoop/CephfsHadoopUnderFileSystem.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+   * @return a new cephfs-hadoop {@link UnderFileSystem} instance
+   */
+  public static CephfsHadoopUnderFileSystem createInstance(AlluxioURI ufsUri,
+      UnderFileSystemConfiguration conf) {
+    Configuration hdfsConf = createConfiguration(conf);
+    return new CephfsHadoopUnderFileSystem(ufsUri, conf, hdfsConf);
+  }
+
+  /**
+   * Constructs a new HDFS {@link UnderFileSystem}.
```
Comment (dbw9580) [inline underfs/cephfs-hadoop/src/main/java/alluxio/underfs/cephfshadoop/CephfsHadoopUnderFileSystem.java:43]:
```suggestion
   * Constructs a new Cephfs-Hadoop {@link UnderFileSystem}.
```

File: underfs/cephfs-hadoop/src/main/java/alluxio/underfs/cephfshadoop/CephfsHadoopUnderFileSystem.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+   *
+   * @param ufsUri the {@link AlluxioURI} for this UFS
+   * @param conf the configuration for this UFS
+   * @param hdfsConf the configuration for HDFS
```
Comment (dbw9580) [inline underfs/cephfs-hadoop/src/main/java/alluxio/underfs/cephfshadoop/CephfsHadoopUnderFileSystem.java:47]:
```suggestion
   * @param hdfsConf the configuration for Cephfs-Hadoop
```

File: underfs/cosn/src/main/java/alluxio/underfs/cosn/CosnUnderFileSystem.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+import alluxio.underfs.UnderFileSystem;
+import alluxio.underfs.UnderFileSystemConfiguration;
+import alluxio.underfs.hdfs.HdfsUnderFileSystem;
+
+import org.apache.hadoop.conf.Configuration;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * HDFS {@link UnderFileSystem} implementation.
```
Comment (dbw9580) [inline underfs/cosn/src/main/java/alluxio/underfs/cosn/CosnUnderFileSystem.java:24]:
```suggestion
 * COSN {@link UnderFileSystem} implementation.
```

File: underfs/ozone/src/main/java/alluxio/underfs/ozone/OzoneUnderFileSystem.java [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+import alluxio.underfs.UnderFileSystem;
+import alluxio.underfs.UnderFileSystemConfiguration;
+import alluxio.underfs.hdfs.HdfsUnderFileSystem;
+
+import org.apache.hadoop.conf.Configuration;
+
+import javax.annotation.concurrent.ThreadSafe;
+
+/**
+ * HDFS {@link UnderFileSystem} implementation.
```
Comment (dbw9580) [inline underfs/ozone/src/main/java/alluxio/underfs/ozone/OzoneUnderFileSystem.java:24]:
```suggestion
 * Ozone {@link UnderFileSystem} implementation.
```

File: underfs/ozone/src/main/java/alluxio/underfs/ozone/OzoneUnderFileSystem.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+    Configuration hdfsConf = createConfiguration(conf);
+    return new OzoneUnderFileSystem(ufsUri, conf, hdfsConf);
+  }
+
+  /**
+   * Constructs a new HDFS {@link UnderFileSystem}.
+   *
+   * @param ufsUri the {@link AlluxioURI} for this UFS
+   * @param conf the configuration for this UFS
+   * @param hdfsConf the configuration for HDFS
```
Comment (dbw9580) [inline underfs/ozone/src/main/java/alluxio/underfs/ozone/OzoneUnderFileSystem.java:47]:
```suggestion
   * @param hdfsConf the configuration for Ozone
```

File: underfs/ozone/src/main/java/alluxio/underfs/ozone/OzoneUnderFileSystemFactory.java [hunk 10]
```diff
@@ -11,24 +11,33 @@
...
+
 import java.util.List;
 import javax.annotation.concurrent.ThreadSafe;
 
 /**
- * Factory for creating {@link HdfsUnderFileSystem}.
+ * Factory for creating {@link OzoneUnderFileSystem}.
  *
- * It caches created {@link HdfsUnderFileSystem}s, using the scheme and authority pair as the key.
+ * It caches created {@link OzoneUnderFileSystem}s, using the scheme and authority pair as the key.
```
Comment (dbw9580) [inline underfs/ozone/src/main/java/alluxio/underfs/ozone/OzoneUnderFileSystemFactory.java:30]: remove comment

**Review by dbw9580 — CHANGES_REQUESTED** [review]
Thanks for the improvement. Mostly LGTM, only a few comments related to javadoc.

**Comment by maobaolong** [comment]
@dbw9580 Thank you for your previous review, I addressed all your comments, PTAL

File: underfs/cephfs-hadoop/src/main/java/alluxio/underfs/cephfshadoop/CephfsHadoopUnderFileSystemFactory.java (hunk 1 above)
Comment (maobaolong) [reply underfs/cephfs-hadoop/src/main/java/alluxio/underfs/cephfshadoop/CephfsHadoopUnderFileSystemFactory.java]: Remove it.

File: underfs/cosn/src/main/java/alluxio/underfs/cosn/CosNUnderFileSystemFactory.java (hunk 2 above)
Comment (maobaolong) [reply underfs/cosn/src/main/java/alluxio/underfs/cosn/CosNUnderFileSystemFactory.java]: Remove it

File: underfs/ozone/src/main/java/alluxio/underfs/ozone/OzoneUnderFileSystem.java [hunk 11]
```diff
@@ -0,0 +1,58 @@
...
+   * @return a new Ozone {@link UnderFileSystem} instance
+   */
+  public static OzoneUnderFileSystem createInstance(AlluxioURI ufsUri,
+      UnderFileSystemConfiguration conf) {
+    Configuration hdfsConf = createConfiguration(conf);
+    return new OzoneUnderFileSystem(ufsUri, conf, hdfsConf);
+  }
+
+  /**
+   * Constructs a new HDFS {@link UnderFileSystem}.
```
Comment (dbw9580) [inline underfs/ozone/src/main/java/alluxio/underfs/ozone/OzoneUnderFileSystem.java:43]:
```suggestion
   * Constructs a new Ozone {@link UnderFileSystem}.
```

**Review by dbw9580 — APPROVED** [review]
LGTM

**Comment by jiacheliu3** [comment]
alluxio-bot, merge this please

**Pull request merged**